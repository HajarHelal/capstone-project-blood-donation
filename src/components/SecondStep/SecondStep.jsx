import React, { useState } from 'react';
import { PropTypes } from 'prop-types';
import { useTranslation } from 'react-i18next';
import Button from '../Button';
import RadioButton from '../RadioButton';
import HeaderPharagrph from '../Paragraph';
import SubHeader from '../SubHeader';
import Donateparagraph from '../Donateparagraph';
import Progress from '../Progress';
import InputSelect from '../InputSelect';

function SecondStep({ donor, onChange, setStep, chooseBlood, city }) {
  const [errorcity, setErrorcity] = useState(false);
  const [errorblood, seterrorblood] = useState(false);
  const { t } = useTranslation();
  const onClickNext = () => {
    if (donor.city !== '' && donor.bloodType !== '') {
      setStep(3);
    }
    if (donor.city === '') {
      setErrorcity(true);
    } else {
      setErrorcity(false);
    }
    if (donor.bloodType === '') {
      seterrorblood(true);
    } else {
      seterrorblood(false);
    }
  };
  const onClickPrevious = () => {
    setStep(1);
  };
  const bloodTypes = Array.from(t('bloodTypes', { returnObjects: true }));
  const cities = Array.from(t('cities', { returnObjects: true }));
  return (
    <div>
      <div className="text-primary-100  sm:text-xl md:text-2xl lg:text-3xl pt-1 font-semibold mb-6 ">
        {t('donate_second_page.great')}
      </div>
      <div className=" w-11/12 xl:w-10/12 mb-6 sm:mb-8 md:mb-12">
        <HeaderPharagrph
          headerText={t('donate_second_page.title')}
          headerFlag={false}
          paragraphText={t('donate_second_page.paragrph')}
          paragraphFlag={false}
        />
      </div>
      <div className="p-6 sm:p-8 md:p-10 border-gray-300 shadow-xl border-2 rounded">
        <Progress steps={2} />
        <SubHeader text={t('donate_second_page.subHeader1')} />
        <Donateparagraph text={t('donate_second_page.subparagrph')} />
        <SubHeader text={t('donate_second_page.subHeader2')} />
        <RadioButton
          donor={donor}
          onChange={onChange}
          radiotext={t('donate_second_page.male_text')}
          RadioOptions={[
            t('donate_second_page.option_male'),
            t('donate_second_page.option_female'),
          ]}
          name="gender"
        />
        <div className="text-sm sm:text-base md:text-lg font-medium inline-block mb-4 sm:mb-6">
          {t('donate_second_page.blood_type_text')}
        </div>
        <div className="flex ml-4 flex-wrap">
          {bloodTypes.map((blood) => (
            <button
              type="button"
              key={blood}
              className={`${
                donor.bloodType === blood ? 'bg-primary text-white' : ''
              } py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 rounded-none border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg`}
              onClick={() => chooseBlood(blood)}
            >
              {blood}
            </button>
          ))}
        </div>
        <div className="mb-8 text:sm text-red-600">
          {errorblood ? t('donate_second_page.errorBlood') : ''}
        </div>
        <div className="w-8/12 sm:w-6/12 lg:w-4/12">
          <InputSelect
            city={city}
            onChange={onChange}
            name="city"
            text={t('donate_second_page.city')}
            options={cities}
            classflag={false}
          />
        </div>
        <div className="mb-8 text:sm text-red-600">
          {errorcity ? t('donate_second_page.errorCity') : ''}
        </div>
        <div className="flex button_flex items-center mb-8">
          <div className="mx-4 sm:mx-8 text-center">
            <Button
              text={t('donate_second_page.button_previous')}
              onclick={onClickPrevious}
              classFlag
            />
          </div>
          <div>
            <Button
              text={t('donate_second_page.button_next')}
              onclick={onClickNext}
              classFlag={false}
            />
          </div>
        </div>
      </div>
    </div>
  );
}
SecondStep.propTypes = {
  donor: PropTypes.object.isRequired,
  city: PropTypes.string.isRequired,
  setStep: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  chooseBlood: PropTypes.func.isRequired,
};
export default SecondStep;
