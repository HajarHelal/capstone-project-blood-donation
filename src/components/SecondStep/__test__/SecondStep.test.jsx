import React from 'react';
import renderer from 'react-test-renderer';
import SecondStep from '../SecondStep';

// This is an example of a Snapshot. It will create a new snapshot which all future
// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer
    .create(
      <SecondStep
        donor={{
          name: '',
          email: '',
          phone: '',
          date: '',
          weight: '',
          sickness: '',
          bloodType: '',
          gender: '',
        }}
        onChange={() => console.log('donate')}
        setStep={() => console.log('donate')}
        chooseBlood={() => console.log('donate')}
        city="Sanaa"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
