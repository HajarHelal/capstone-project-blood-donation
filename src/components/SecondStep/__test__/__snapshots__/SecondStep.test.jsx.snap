// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="text-primary-100  sm:text-xl md:text-2xl lg:text-3xl pt-1 font-semibold mb-6 "
  >
    donate_second_page.great
  </div>
  <div
    className=" w-11/12 xl:w-10/12 mb-6 sm:mb-8 md:mb-12"
  >
    <div>
      <h1
        className="text-primary-100  sm:text-xl md:text-2xl lg:text-3xl pt-1 font-semibold  "
      >
        donate_second_page.title
      </h1>
      <p
        className=" pt-3 pb-5  sm:text-sm md:text-base lg:text-lg   "
      >
        donate_second_page.paragrph
      </p>
    </div>
  </div>
  <div
    className="p-6 sm:p-8 md:p-10 border-gray-300 shadow-xl border-2 rounded"
  >
    <div
      className="flex  items-center w-10/12  mb-10"
    >
      <div
        className="text-base md:text-lg lg:text-xl font-semibold  mx-4 sm:mx-8"
      >
        donate_first_page.step
        2 / 3
      </div>
      <div
        className="flex-1 h-6 sm:h-8 md:h-10 bg-gray-100"
      >
        <div
          className="w-4/12 h-6 sm:h-8 md:h-10 hidden bg-primary"
        />
        <div
          className="w-8/12 h-6 sm:h-8 md:h-10  bg-primary"
        />
        <div
          className="w-100 h-6 sm:h-8 md:h-10 hidden bg-primary"
        />
      </div>
    </div>
    <div
      className="text-base md:text-lg lg:text-xl font-medium mb-6"
    >
      donate_second_page.subHeader1
    </div>
    <div
      className="text-sm sm:text-base md:text-lg  mb-10 leading-relaxed w-10/12"
    >
      donate_second_page.subparagrph
    </div>
    <div
      className="text-base md:text-lg lg:text-xl font-medium mb-6"
    >
      donate_second_page.subHeader2
    </div>
    <div
      className="block mb-8"
      name="gender"
      onChange={[Function]}
      value=""
    >
      <span
        className="text-sm sm:text-base md:text-lg font-medium inline-block mb-2"
      >
        donate_second_page.male_text
      </span>
      <div
        className="mt-2"
      >
        <label
          className="inline-flex items-center"
          htmlFor="sa"
        >
          <input
            className="form-radio text-green-500 h-6 w-6"
            defaultChecked={false}
            name="gender"
            type="radio"
            value="donate_second_page.option_male"
          />
          <span
            className="mx-4"
          >
            donate_second_page.option_male
          </span>
        </label>
      </div>
      <div
        className="mt-2"
      >
        <label
          className="inline-flex items-center"
          htmlFor="sa"
        >
          <input
            className="form-radio text-green-500 h-6 w-6"
            defaultChecked={false}
            name="gender"
            type="radio"
            value="donate_second_page.option_female"
          />
          <span
            className="mx-4"
          >
            donate_second_page.option_female
          </span>
        </label>
      </div>
    </div>
    <div
      className="text-sm sm:text-base md:text-lg font-medium inline-block mb-4 sm:mb-6"
    >
      donate_second_page.blood_type_text
    </div>
    <div
      className="flex ml-4 flex-wrap mb-8"
    >
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        b
      </button>
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        l
      </button>
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        o
      </button>
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        o
      </button>
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        d
      </button>
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        T
      </button>
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        y
      </button>
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        p
      </button>
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        e
      </button>
      <button
        className=" py-2 px-2 sm:py-2 sm:px-4 lg:py-4 lg:px-8 border-2 border-gray-400 hover:bg-primary hover:text-white hover:border-primary text-sm sm:textt-base md:text-lg"
        onClick={[Function]}
        type="button"
      >
        s
      </button>
    </div>
    <div
      className="w-8/12 sm:w-6/12 lg:w-4/12"
    >
      <div
        className="mb-4 sm:mb-6"
      >
        <label
          className="block"
          htmlFor="nowhere"
        >
          <div
            className="text-sm sm:text-base md:text-lg font-medium inline-block mb-4 sm:mb-6"
          >
            donate_second_page.city
          </div>
          <select
            className="form-select block w-full border-gray-300 border-2 py-2 px-4 text-sm sm:text-base md:text-lg "
            name="city"
            onBlur={[Function]}
            onChange={[Function]}
            value="Sanaa"
          >
            <option>
              c
            </option>
            <option>
              i
            </option>
            <option>
              t
            </option>
            <option>
              i
            </option>
            <option>
              e
            </option>
            <option>
              s
            </option>
          </select>
        </label>
      </div>
    </div>
    <div
      className="flex  mb-8"
    >
      <div
        className="mx-4 sm:mx-8 text-center"
      >
        <button
          className="inline-block text-primary-100 text-base md:text-lg  text-center md:font-medium bg-white  md:mt-8 mt-4 px-8 md:px-12 lg:px-16 py-1 lg:py-2  shadow-md transition duration-500 ease-in-out transform hover:scale-95"
          onClick={[Function]}
          type="button"
        >
          donate_second_page.button_previous
        </button>
      </div>
      <div>
        <button
          className="inline-block text-white text-base md:text-lg  text-center md:font-medium bg-primary  md:mt-8 mt-4 px-8 md:px-12 lg:px-16 py-1 lg:py-2  shadow-md transition duration-500 ease-in-out transform hover:scale-95"
          onClick={[Function]}
          type="button"
        >
          donate_second_page.button_next
        </button>
      </div>
    </div>
  </div>
</div>
`;
